app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result_dict = {}
+
+    def inner(*args, **kwargs) -> int:
```

Comment:
The return type hint for the `inner` function is specified as `int`, which may not be accurate. Since the `inner` function returns the result of `func`, the return type should match the return type of `func`. Consider using `Any` or the same return type as `func` if known.